File: app/src/main/java/org/witness/proofmode/OnBootReceiver.java
package org.witness.proofmode;

import static org.witness.proofmode.ProofMode.PREFS_DOPROOF;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Build;
import android.preference.PreferenceManager;

import org.witness.proofmode.ProofModeApp;

/**
 * Created by n8fr8 on 10/10/16.
 */
public class OnBootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {

        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);

        if (prefs.getBoolean(PREFS_DOPROOF, false)) {


            Intent intentService = new Intent(context, ProofService.class);

            if (Build.VERSION.SDK_INT >= 26) {
                context.startForegroundService(intentService);

            } else {
                context.startService(intentService);
            }

        }
    }
}

File: app/src/main/java/org/witness/proofmode/OnBootReceiver.kt
package org.witness.proofmode

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import androidx.preference.PreferenceManager

/**
 * Created by n8fr8 on 10/10/16.
 */
class OnBootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val prefs = PreferenceManager.getDefaultSharedPreferences(context)
        if (prefs.getBoolean(ProofMode.PREFS_DOPROOF, false)) {

            ProofMode.initBackgroundService(context)

            /**
            val intentService = Intent(context, ProofService::class.java)
            if (Build.VERSION.SDK_INT >= 26) {
                context.startForegroundService(intentService)
            } else {
                context.startService(intentService)
            }**/
        }
    }
}